ibis/backends/sql/compilers/duckdb.py

```diff
@@ -177,16 +177,19 @@ def visit_ArrayRepeat(self, op, *, arg, times):
         return self.f.flatten(self.f.list_apply(self.f.range(times), func))
 
     def visit_ArraySlice(self, op, *, arg, start, stop):
+        import duckdb
+
         arg_length = self.f.len(arg)
 
         if start is None:
             start = 0
-        else:
-            start = self.f.least(arg_length, self._neg_idx_to_pos(arg, start))
-
         if stop is None:
             stop = arg_length
-        else:
+
+        # TODO: once we only support duckdb >= 0.9.0, we can remove this
+        if duckdb.__version__ < "0.9.0":
```

Comment:
We should just bump to 0.9.0 as a lower bound for the next release, then we don't have to futz with with this.

I'll update your PR to make these changes.